Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/format/xml/XLink.java

```diff
@@ -0,0 +1,576 @@
+     *            link title(s) as human readable text
+     * @param actuate
+     *            when to load the resource
+     * @param show
+     *            context to show the resource in
+     * @return the created link
+     */
+    public static Node createSimpleLink(Storage storage, Optional<IdentifiableNode> relation, IdentifiableNode to,
+            Collection<LangString> title, Optional<Actuate> actuate, Optional<Show> show) {
+
+        return createSimpleLink(storage, relation, to.getIdentifier(),
+            to instanceof Node ? ((Node) to).get(RDF.TYPE).leaves() : Collections.emptySet(), title, actuate, show);
+    }
+
+    /**
+     * Creates a simple link.
+     *
+     * @param storage
+     *            storage instance to create the link in
+     * @param relation
+     *            Relation description. For example:
+     *            {@code http://purl.org/dc/terms/references}
+     * @param to
+     *            link destination
+     * @param type
+     *            Collection of {@code rdf:type}s of the linked resource
+     * @param title
+     *            link title(s) as human readable text
+     * @param actuate
+     *            when to load the resource
+     * @param show
+     *            context to show the resource in
+     * @return the created link
+     */
+    public static Node createSimpleLink(Storage storage, Optional<IdentifiableNode> relation, String to,
+            Collection<String> type, Collection<LangString> title, Optional<Actuate> actuate, Optional<Show> show) {
+        Node node = storage.createNode();
+        node.put(HREF, to);
+        type.forEach(entry -> node.put(ROLE, entry));
+        relation.ifPresent(value -> node.put(ARCROLE, value.getIdentifier()));
+        if (title.size() == 1) {
+            LangString entry = title.iterator().next();
+            node.put(TITLE, entry.getValue());
+            node.put(XML.LANG, entry.getLanguageTag());
+        } else {
+            title.forEach(entry -> node
+                    .add(storage.createNode(TITLE).put(XML.LANG, entry.getLanguageTag()).setValue(entry.getValue())));
+        }
+        actuate.ifPresent(value -> node.put(ACTUATE, value.toString()));
+        show.ifPresent(value -> node.put(SHOW, value.toString()));
+        return node;
+    }
+
+    private String identifier;
+
+    /**
+     * Enum constants constructor.
+     *
+     * @param identifier
+     *            referenced URL
+     */
+    private XLink(String identifier) {
+        this.identifier = identifier;
+    }
+
+    @Override
+    public String getIdentifier() {
+        return identifier;
+    }
+
+    @Override
+    public RDFNode toRDFNode(Model model, Boolean unused) {
+        return model.createResource(identifier);
+    }
+
+    /**
+     * Returns a version of this node reference which, in a debugger, will
+     * symbolically represent it.
+     */
+    @Override
+    public String toString() {
+        return '↗' + identifier;
+    }
+
```

Comment:
General remark to the file, it would be more readable if it would be sorted in some way, for e.g. first attributes, next inner enums, at the end methods.

Reply:
It is sorted:
1. enum constants, in alphabetical order
2. inner enums, in alphabetical order
3. (normal) constants, in alphabetical order
4. static methods, in alphabetical order
5. constructor, in alphabetical order
6. non-static methods, in alphabetical order

This is the order Eclipse provides me with when I use the "save action" to “sort all members”.